tensorflow/contrib/cmake/tf_shared_lib.cmake

```diff
@@ -95,10 +95,18 @@ if(WIN32)
   add_dependencies(tensorflow tensorflow_static)
 endif(WIN32)
 
-install(TARGETS tensorflow
+target_include_directories(tensorflow PUBLIC 
+    $<INSTALL_INTERFACE:include/>
+    $<INSTALL_INTERFACE:include/external/nsync/public>)
```

Comment:
Are the nsync headers missing a corresponding `install(...)` statement somewhere below?

Reply:
The headers are needed within _core/platform/default/mutex.h_, which will be installed.
The nsync headers are installed, but the tensorflow-target needs the ${install_prefix}/include/external/nsync/public as a target include directories.

We can change the path in the mutex.h file or adding the installed folder to the target within _install\_interface_

Reply:
Ack, that sounds good (I'm not particularly clear on how CMake installation works wrt. external packages). If it works for you, SGTM!

Reply:
This might not be the only target dependent installation property. Maybe the exported libraries will also need some target compile definitions. But these can easily added